internal/venice-common/src/main/java/com/linkedin/venice/blobtransfer/BlobTransferManager.java

```diff
@@ -0,0 +1,37 @@
+package com.linkedin.venice.blobtransfer;
+
+import com.linkedin.venice.exceptions.VeniceResourceNotFoundException;
+import java.io.InputStream;
+import java.util.concurrent.CompletionStage;
+
+
+/**
+ *
+ * A BlobTransferManager is responsible for transferring blobs between two entities, either that Peer to Peer or node to
+ * a blob store and vice versa. The underlying blob client is responsible for the actual transfer of the blob.
+ * @param <T> the type of the object from the underlying blob client to indicate the upload status. It can be a blob ID
+ *           indicating the blob has been uploaded or an enum representing the status of the blob transfer.
+ */
+public interface BlobTransferManager<T> extends AutoCloseable {
+  /**
+   * Get the blobs for the given storeName and partition
+   * @param storeName
+   * @param partition
+   * @return the InputStream of the blob
+   * @throws VeniceResourceNotFoundException
+   */
+  CompletionStage<InputStream> get(String storeName, int partition) throws VeniceResourceNotFoundException;
```

Comment:
`InputStream` is an abstract class. Is this what we want here? If there is a need for different blob implementations to provide different types of input streams, should we have an additional generic type in the interface, e.g. `I extends InputStream`? Or should we constrain it to a specific type, e.g. a `BufferedInputStream`?

Reply:
This's still a TBD honestly. I borrowed the idea from Ambry since it's simple to start with.
I don't have enough information to determine whether we need generic type (does it worth extra code complexity) or use a specific type (`BufferedInputStream`).  The chance is we may use some customized class in the future for optimization.

Reply:
I think I'm leaning to `BufferedInputStream` now since it should be sufficient for quite some time

Reply:
Updated.

Reply:
Isn't `InputStream` better in the sense the underlying interactions with concrete blob service may return an `InputStream` which isn't buffered and keeping it as-is is more flexible?

Agree, that we should do `CompletionStage<? extends InputStream>` instead of `CompletionStage<InputStream>`

Reply:
Oh I missed this comment. 
> concrete blob service may return an InputStream which isn't buffered 

For P2P large file transfer, I think input stream should always be buffered or similarly optimized for better performance and Ambry does something similar but they do only return `InputStream` for simplicity reason I guess.
I don't have strong preference on the generic type but would using wildcard be safe here? I don't feel the casting is safe here especially for `inputStream`

Reply:
there is still type restrictions imposed with the `? extends InputStream` just that we aren't explicitly taking a type through interface definition. You probably don't require any casting.